exist-core/src/main/java/org/exist/xquery/functions/securitymanager/IdFunction.java

```diff
@@ -77,13 +77,18 @@ private org.exist.dom.memtree.DocumentImpl functionId() {
 
             builder.startElement(new QName("id", SecurityManagerModule.NAMESPACE_URI, SecurityManagerModule.PREFIX), null);
 
-            builder.startElement(new QName("real", SecurityManagerModule.NAMESPACE_URI, SecurityManagerModule.PREFIX), null);
-            subjectToXml(builder, context.getRealUser());
-            builder.endElement();
+            final Subject realUser = context.getRealUser();
+            if (realUser != null) {
```

Comment:
realUser will never be null

Reply:
see comment above

Reply:
throw XPathException in this case?

Reply:
A null value should not happen but... defensive programming in a complex application is not a too bad idea I'd say.

A log to detect the situation/throw exception ... both would be OK for me.